File: project-service/src/main/java/io/studyscrum/project/model/Project.java
package io.studyscrum.project.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "projects")
public class Project {

    @Id
    private String id;

    private String name;

    private String description;

    private String courseName;

    private String courseCode;

    private Instant semesterStart;

    private Instant semesterEnd;

    private Instant projectDeadline;

    @Builder.Default
    private List<ProjectMember> members = new ArrayList<>();

    private String status;

    @CreatedDate
    private Instant createdAt;

    @LastModifiedDate
    private Instant updatedAt;
}

File: frontend/src/types/project.ts
export type ProjectStatus = 'ACTIVE' | 'ARCHIVED';

export type MemberRole = 'LEAD' | 'MEMBER';

export interface ProjectMember {
  userId: string;
  email: string;
  name: string;
  role: MemberRole;
  joinedAt: string;
}

export interface Project {
  id: string;
  name: string;
  description?: string;
  courseName?: string;
  courseCode?: string;
  semesterStart?: string;
  semesterEnd?: string;
  projectDeadline?: string;
  members: ProjectMember[];
  status: ProjectStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProjectRequest {
  name: string;
  description?: string;
  courseName?: string;
  courseCode?: string;
  semesterStart?: string;
  semesterEnd?: string;
  projectDeadline?: string;
}
